src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -802,105 +998,215 @@ public static unsafe Vector256<ulong> Create(Vector128<ulong> lower, Vector128<u
         /// <summary>Creates a new <see cref="Vector256{Byte}" /> instance with the first element initialized to the specified value and the remaining elements initialized to zero.</summary>
         /// <param name="value">The value that element 0 will be initialized to.</param>
         /// <returns>A new <see cref="Vector256{Byte}" /> instance with the first element initialized to <paramref name="value" /> and the remaining elements initialized to zero.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe Vector256<byte> CreateScalar(byte value)
         {
-            var result = Vector256<byte>.Zero;
-            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<byte>, byte>(ref result), value);
-            return result;
+            if (Avx.IsSupported)
+            {
+                return Vector128.CreateScalar(value).ToVector256();
```

Comment:
Generates identical code to the V128 versions. The next access of the result is using the `ymm` register.

For example:
```asm
       movzx    rax, dl
       movzx    rax, al
       vmovd    xmm0, xrax
       vmovaps  xmm0, xmm0
       vmovupd  ymmword ptr[rcx], ymm0
```

Reply:
Hmm, `vmovaps  xmm0, xmm0` should not be generated.

Reply:
I thought this a bit. We are doing the wrong things for some `Unsafe` intrinsic. 
For example, `ToVector256` should generate `vmovaps` iff `targetReg != opReg`. 
For `ToVector256Unsafe`, we should just fold it away at the front-end (like `As<T>`), which let the `ToVector256Unsafe`'s operand (XMM) be used as YMM directly. That is what we originally want.

Reply:
> Hmm, vmovaps xmm0, xmm0 should not be generated.

Ah, right; since `vmovd` will already zero the upper bits.

Reply:
> For ToVector256Unsafe, we should just fold it away at the front-end (like As<T>), which let the ToVector256Unsafe's operand (XMM) be used as YMM directly. That is what we originally want.

I'm not sure we can. The original data is a `TYP_SIMD16` but the target data is larger (a `TYP_SIMD32`). If we don't actually resize the data, then a later consumer could end up with an AV (since it is now trying to read a `TYP_SIMD32`, when only `TYP_SIMD16` was valid).

Reply:
> For example, ToVector256 should generate vmovaps iff targetReg != opReg

I don't think we have any guarantee on the value of the upper bits today. The most likely case is that the upper bits will be zero, but there are still a few scenarios around P/Invoke and interop that (I believe) could result in the upper bits being something else.

Reply:
Hmm, seems we don’t need ToVector256Unsafe actually...

Reply:
> Hmm, seems we don’t need ToVector256Unsafe actually..

What do you mean?

We have at least interop scenarios and the case where a user performs a 256-bit operation but only returns `GetLower()` (and possibly others) where this would make a difference. For both of these it results in a case where `ToVector256` must insert a `movaps` but where `ToVector256Unsafe` can avoid it as the upper bits may be non-zero.

Reply:
I suspect that eliding the widening operator could indeed lead to problems. This is something that could perhaps be handled in `Lowering`.

Reply:
> GetLower() (and possibly others) where this would make a difference. For both of these it results in a case where ToVector256 must insert a movaps but where ToVector256Unsafe can avoid it as the upper bits may be non-zero.

Ah, right. Thanks